vm/env.go

```diff
@@ -223,9 +226,15 @@ func (e *Env) Define(k string, v interface{}) error {
 	if strings.Contains(k, ".") {
 		return fmt.Errorf("Unknown symbol '%s'", k)
 	}
-	val, ok := v.(reflect.Value)
-	if !ok {
-		val = reflect.ValueOf(v)
+	var val reflect.Value
+	if v == nil {
+		val = NilTypeNonPointer
```

Comment:
What is achieved by this? You can still set nil to variable now. What is the benefit of `NilTypeNonPointer`.

Reply:
This is the smallest change I could find that will return the correct nil value for a variable. If there is a better way, I would really like to know and understand.

Reply:
Could you provide example how it behaved before and with this change?

Reply:
Variables returned would be set to <invalid reflect.Value>instead of <nil>.

Here is one of the tests that would catch it:
{script: "a = b", input: map[string]interface{}{"b": nil}, runOutput: nil, ouput: map[string]interface{}{"a": nil, "b": nil}},

Here is the output from the test when the change is reverted:
outputName b - received: <invalid reflect.Value> expected: <nil> - script: a = b

Reply:
Removed NilTypeNonPointer. Changed NilValue to be an inteface{} (nil) which should fix the issue and not cause this weirdness.